### PR TITLE
OCPBUGS-15087: templates: Introduce kubelet-dependencies.target

### PR DESCRIPTION
**Pull request by cgwalters**
The primary motivation here is to eventually stop pulling container images `Before=network-online.target` because it creates complicated dependency loops.

We introduce a new `kubelet-dependencies.target` - both `crio.service` and `kubelet.service` are `After+Requires=kubelet-dependencies.target`.

This also makes it cleaner for pre-kubelet services, which can just order themselves `Before=kubelet-dependencies.target`.

This is just a small preparatory PR to introduce the target unit. The real bigger change will come when we move services like `ovs-configuration.service` and `node-valid-hostname.service` to actually be `After=network-online.target` + `Before=kubelet-dependendies.target`.

Crucially, this will unblock services like `machine-config-daemon-pull.service` that want to fetch containers *before* `kubelet.service` but want to be `After=network-online.target`.

### CODE REVIEW

**Comment by cgwalters** [comment]
xref https://github.com/openshift/machine-config-operator/pull/3858 and https://github.com/openshift/machine-config-operator/pull/3788 which want this.

**Comment by cgwalters** [comment]
> The primary motivation here is to eventually stop pulling container images Before=network-online.target because it creates complicated dependency loops.

Another way to say this: I don't think e.g. `ovs-configuration.service` needs to be `Before=network-online.target`.  It only needs to be `Before=kubelet.service crio.service`.  Injecting all this stuff into `network-online.target` is just polluting the meaning of the target, particularly because we actually do need to fetch container images before reaching that target!

**Comment by cgwalters** [comment]
Hmm, I don't understand the build controller unit failures?

```
     build_controller_test.go:457: 
        	Error Trace:	/go/src/github.com/openshift/machine-config-operator/pkg/controller/build/fixtures_test.go:181
        	            				/go/src/github.com/openshift/machine-config-operator/pkg/controller/build/build_controller_test.go:457
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestBuildControllerMultipleOptedInPools/Happy_Path_Multiple_Configs/worker/CustomBuildPod/rendered-worker-9
        	Messages:   	MachineConfigPool worker never reached desired state
    build_controller_test.go:463: Config name, actual: rendered-worker-9, expected: rendered-worker-9
    build_controller_test.go:464: Is build success? false
    build_controller_test.go:465: Has all MachineConfig refs? true 
```

Not immediately reproducing this locally...

**Comment by cybertron** [comment]
Some thoughts on this, but the TLDR is that I think we should try it.

I have concerns with reaching network-online before ovs-configuration has run since that will cause a network blip when it runs and reconfigures things. If something does rely on network-online to know when networking is ready, it may start before ovs-configuration runs and then be broken when the bridge changes are made.

However, I think this is mitigated by a couple of things:

1.  No user workloads will be running until Kubelet starts, so we should have the ability to massage dependencies in any services deployed prior to that. I guess the exception to this would be if a user deployed a service directly to the system via machine-config. If such a service got broken by this change, at least it's simple to fix (change target from network-online to kubelet-dependencies, which should be largely equivalent).
2. The bugs being caused by this[0] are severe enough that we may need to accept some risk of breaking edge cases to fix the more common case. Being unable to login to a system to debug a problem during early boot makes troubleshooting very difficult.

0: https://issues.redhat.com/browse/OCPBUGS-18541

**Comment by cybertron** [comment]
I did some testing with this locally and it seems to do what we want. I updated the ovs-configuration and node-valid-hostname dependencies to use this target and then injected an artificial delay into nodeip-configuration. It no longer blocked network-online:

```
[root@master-0 core]# systemctl status nodeip-configuration | grep Active
     Active: inactive (dead) since Fri 2023-09-22 22:41:45 UTC; 29min ago
[root@master-0 core]# systemctl status network-online.target | grep Active
     Active: active since Fri 2023-09-22 22:40:44 UTC; 31min ago # <-- Earlier than nodeip-configuration finished
[root@master-0 core]# systemctl status kubelet | grep Active
     Active: active (running) since Fri 2023-09-22 22:41:56 UTC; 30min ago # <-- Still after nodeip-configuration finished
```

**Comment by cgwalters** [comment]
> I have concerns with reaching network-online before ovs-configuration has run since that will cause a network blip when it runs and reconfigures things. If something does rely on network-online to know when networking is ready, it may start before ovs-configuration runs and then be broken when the bridge changes are made.

The other factor is that `network-online.target` is really just a hint.  Anything doing networking *must* be resilient to blips and flakes.  Having things poll to eventual consistentency is the whole model we're aiming for in Kubernetes.

(It's really just this "pre-kubelet" stuff where we haven't really ingrained that mindset!)

**Comment by cgwalters** [comment]
OK, I've verified that a 4.6 node can join the cluster with this, and that there are no ordering cycles emitted at any point.

File: templates/common/_base/units/machine-config-daemon-firstboot.service.yaml
Comment (cybertron) [inline templates/common/_base/units/machine-config-daemon-firstboot.service.yaml:11]: I think we want to keep ovs-configuration here, unless that dependency is just no longer needed for some other reason.

Comment (cgwalters) [reply templates/common/_base/units/machine-config-daemon-firstboot.service.yaml:11]: Currently they're effectively exclusive I believe because of the `ConditionPathExists=!/etc/ignition-machine-config-encapsulated.json`.

Comment (cgwalters) [reply templates/common/_base/units/machine-config-daemon-firstboot.service.yaml:11]: That said...I think the dual-root upgrade model may push us away from the encapsulated stamp file.

So...a common systemd thing is to have a `foo.target` and `foo-pre.target`.  We could make all the kubelet deps be `After=kubelet-dependencies-pre.target`, and then stick `machine-config-daemon-firstboot.service` under `Before=kubelet-dependencies-pre.target` or so.

I don't think we need to block this PR on that though?

**Comment by cgwalters** [comment]
This is a larger rework spawned out of https://github.com/openshift/machine-config-operator/pull/3788
@sergiordlr mind testing this one?  Thanks!

**Review by yuqi-zhang — APPROVED** [review]
Makes sense to me. cc @rphillips to see if there are any concerns with ordering this from the node side

**Comment by sergiordlr** [comment]
Verified using IPI on OSP.

1. Install an OSP 4.15 cluster including this fix
2. Create a new Machineset using an old 4.6 cloud image
3. Scale the new Machineset several times.

We created 25 new workers aprox. using the 4.6 cloud image. All clusters were created without problems.

No ordering cycle problems in the journal of the new nodes either:
`
sh-5.1# journalctl -o with-unit  |grep "ordering cycle"
`

We can add the qe-approved label.

Thank you very much!!

/label qe-approved

**Comment by cgwalters** [comment]
OK cool, I think this just needs a lgtm now!

**Comment by cybertron** [comment]
/lgtm

Thanks!

File: templates/common/_base/units/machine-config-daemon-firstboot.service.yaml
Comment (sinnykumari) [reply templates/common/_base/units/machine-config-daemon-firstboot.service.yaml:11]: wanted to double check. deps to ovs-configuration.service was added in recent PR https://github.com/openshift/machine-config-operator/pull/3858/files#diff-3380ef05804fcff21705fcca487ee27355893c239fb1f0df5077f9dc90cb6719R11 to keep things working when we skip reboot during initial pivot. Is this covered?

Comment (cgwalters) [reply templates/common/_base/units/machine-config-daemon-firstboot.service.yaml:11]: OK yes you're right, this change will re-break that.

**Comment by cgwalters** [comment]
/lgtm cancel per
https://github.com/openshift/machine-config-operator/pull/3865#discussion_r1338688413

**Comment by cgwalters** [comment]
/lgtm cancel
per https://github.com/openshift/machine-config-operator/pull/3865#discussion_r1338688413

**Comment by cgwalters** [comment]
OK there's really no small band aid for this; the tech debt just needed to be addressed.  I've added [another commit](https://github.com/openshift/machine-config-operator/pull/3865/commits/afc666d34cde100c0721e83bb919c204d9499004) which further cleans things up, but this needs more testing.

**Comment by cgwalters** [comment]
There is no CI covering the https://github.com/openshift/machine-config-operator/pull/3858/files#diff-3380ef05804fcff21705fcca487ee27355893c239fb1f0df5077f9dc90cb6719R11 path yet, right/

**Comment by sergiordlr** [comment]
My OSP installation is failing with the following error


 ```
$ oc get mcp -oyaml master
.........
 - lastTransitionTime: "2023-09-27T17:16:07Z"
    message: 'Node sregidor415-xxqcv-master-0 is reporting: "unexpected on-disk state
      validating against rendered-master-6758e1bbb546c6ce4dbfa4c6dc674222: missing
      expected kernel arguments: [systemd.unified_cgroup_hierarchy=1 cgroup_no_v1=\"all\"
      psi=1]", Node sregidor415-xxqcv-master-1 is reporting: "unexpected on-disk state
      validating against rendered-master-6758e1bbb546c6ce4dbfa4c6dc674222: missing
      expected kernel arguments: [systemd.unified_cgroup_hierarchy=1 cgroup_no_v1=\"all\"
      psi=1]", Node sregidor415-xxqcv-master-2 is reporting: "unexpected on-disk state
      validating against rendered-master-6758e1bbb546c6ce4dbfa4c6dc674222: missing
      expected kernel arguments: [systemd.unified_cgroup_hierarchy=1 cgroup_no_v1=\"all\"
      psi=1]"'
    reason: 3 nodes are reporting degraded status on sync
    status: "True"
    type: NodeDegraded


```
It is the same error that we can observe in the failed prow jobs, for example this one from https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3865/pull-ci-openshift-machine-config-operator-master-e2e-gcp-op/1707045822169878528/artifacts/e2e-gcp-op/gather-must-gather/artifacts/

So it is not necessary that I add the mustgather info, since it is already in the prow jobs. (If you need it anyway, please, tell me and I'll re-execute it)

```
$ omg get mcp -o yaml master
.................
  - lastTransitionTime: '2023-09-27T15:41:55Z'
    message: 'Node ci-op-z96rrmzx-1354f-mscwl-master-0 is reporting: "unexpected on-disk
      state validating against rendered-master-60e6de5317edc5eb3af8f13efe60852a: missing
      expected kernel arguments: [systemd.unified_cgroup_hierarchy=1 cgroup_no_v1=\"all\"
      psi=1]", Node ci-op-z96rrmzx-1354f-mscwl-master-1 is reporting: "unexpected
      on-disk state validating against rendered-master-60e6de5317edc5eb3af8f13efe60852a:
      missing expected kernel arguments: [systemd.unified_cgroup_hierarchy=1 cgroup_no_v1=\"all\"
      psi=1]", Node ci-op-z96rrmzx-1354f-mscwl-master-2 is reporting: "unexpected
      on-disk state validating against rendered-master-60e6de5317edc5eb3af8f13efe60852a:
      missing expected kernel arguments: [systemd.unified_cgroup_hierarchy=1 cgroup_no_v1=\"all\"
      psi=1]"'

```

**Comment by cgwalters** [comment]
Yep will debug and then move out of draft when ready

**Comment by sinnykumari** [comment]
> There is no CI covering the https://github.com/openshift/machine-config-operator/pull/3858/files#diff-3380ef05804fcff21705fcca487ee27355893c239fb1f0df5077f9dc90cb6719R11 path yet, right?

Unfortunately no. I think once we have https://github.com/openshift/os/pull/657 , it will be easier to test.

**Comment by cgwalters** [comment]
OK, fixed the [silly omission](https://github.com/openshift/machine-config-operator/compare/c0a3d1804238e23c4b83120b0ca8b9f4709e697b..5f90b8d3942d336d17981a7acbbd943466924ccb) that caused the install failures, and now we have a green e2e-aws-ovn.

@sergiordlr this could now do with some retesting, I'll be doing the same.  :bow:

**Comment by sergiordlr** [comment]
Current testing status

We have installed it in
- UPI on azure
- IPI on Alibaba (alibaba did not install because we hit this issue: https://issues.redhat.com/browse/OCPBUGS-17652, but it is not related to our change, so we ignored it)

Using IPI on OSP
1. Create a 4.15 cluster
2. Use the 4.6 image to create a machineset
3. We created 25 new workers aprox. No problem happened.
4. No ordering cycle problems `journalctl -o with-unit  |grep "ordering cycle"`



Using IPI on vsphere
1. Create a 4.15 clutser
2. Use the 4.6 image to create a machineset
3. We created 7 new workers aprox. No problem found.
4. No ordering cycle problems `journalctl -o with-unit  |grep "ordering cycle"`

In the IPI on vsphere execution did not run the `nodeip-configuration-vsphere-upi.service` service. Since the last comment in the jira ticket it is important that we test that this service has run, and that's the reason why we don't add the qe-approved label yet. 

I'm trying to find an installation template that triggers this service in vsphere.

**Comment by sergiordlr** [comment]
The upi on vsphere using worker. We could add the new nodes.
The nodeip-configuration-vsphere-upi.service was executed after rhcos-growpart.service.

New changes have been done. We will re-execute again the osp and vsphere tests before adding the qe-approved label.

**Comment by sergiordlr** [comment]
With the latest changes my OSP installation is failing. Only one master reached the "kubelet-dependencies.target", the other 2 didn't. We should see the same problem in the prow jobs. 

I added a comment in the jira ticket linking to the journal log of a master node that could not join to the cluster.

**Comment by cgwalters** [comment]
```
Thu 2023-09-28 14:36:05 UTC sregidor715-pzx9g-master-0 init.scope[1]: machine-config-daemon-pull.service: Found ordering cycle on nodeip-configuration.service/start
Thu 2023-09-28 14:36:05 UTC sregidor715-pzx9g-master-0 init.scope[1]: machine-config-daemon-pull.service: Found dependency on firstboot-osupdate.target/start
Thu 2023-09-28 14:36:05 UTC sregidor715-pzx9g-master-0 init.scope[1]: machine-config-daemon-pull.service: Found dependency on machine-config-daemon-firstboot.service/start
Thu 2023-09-28 14:36:05 UTC sregidor715-pzx9g-master-0 init.scope[1]: machine-config-daemon-pull.service: Found dependency on machine-config-daemon-pull.service/start
Thu 2023-09-28 14:36:05 UTC sregidor715-pzx9g-master-0 init.scope[1]: machine-config-daemon-pull.service: Job nodeip-configuration.service/start deleted to break ordering cycle starting with machine-config-daemon-pull.service/start
```

Looking

**Comment by cgwalters** [comment]
OK, the e2e-aws-ovn run passed install, so let's go ahead and run all the rest of the default presubmits.

**Comment by cgwalters** [comment]
That e2e-aws-ovn failure really can't be related...
/test e2e-aws-ovn

**Comment by cgwalters** [comment]
Dunno about the hypershift failures; we had a green on earlier runs (albeit with fewer changes).  I don't yet have expertise in debugging those failures like I have (painfully) built up for standalone :cry:

**Comment by cgwalters** [comment]
Just a note: The previous payload tests on the smaller version of this PR were all green except one, for reasons that don't look obviously related to me.

**Comment by sergiordlr** [comment]
Current testing status:

We installed the fix in azure (functionality-testing/aos-4_15/upi-on-azure/versioned-installer). It was OK.

In OSP (private-templates/functionality-testing/aos-4_15/ipi-on-osp/versioned-installer)   we executed the following steps:
1. Install 4.15 + current fix
2. Create a new machineset using a 4.6 cloud image
3. Create 20+ new worker nodes using the machineset created in step 2
4. No ordering cycle problems `journalctl -o with-unit  |grep "ordering cycle"`

We are having problems with the installation of vsphere clusters. Currently all installations are failing in our vsphere environment (no matter the version). Hence, we have not been able to test this fix in vsphere.

Since vsphere testing it is important  because of the `nodeip-configuration-vsphere-upi.service`, we will not add the qe-approved label until we can test this environment. Currently it is PTO in China, so it could take some time until this problem is fixed. I will try to finish the verification as soon as possible.

**Comment by sergiordlr** [comment]
We executed the following steps using UPI on Vsphere

1. Install 4.15 + this fix
2. Create new worker nodes using `rhcos-46.82.202109242004-0-vmware.x86_64.ova` as the cloud image

No ordering cycle problems `journalctl -o with-unit  |grep "ordering cycle"`

`rhcos-growpart.service` was run before `nodeip-configuration-vsphere-upi.service`

```
Tue 2023-10-03 09:02:08 UTC sregidoru315v-2pp9l-worker-0-modified-bs8bq rhcos-growpart.service[1812]: data blocks changed from 727296 to 31321851
.....
Tue 2023-10-03 09:06:18 UTC sregidoru315v-2pp9l-worker-0-modified-bs8bq nodeip-configuration-vsphere-upi.service[1201]: Getting image source signatures

```


We executed the following tests in IPI on AWS

- "[sig-mco] MCO scale Author:sregidor-NonHyperShiftHOST-NonPreRelease-Longduration-Medium-64623-Machine Config Server CA rotation. IPI. [Disruptive] [Serial]" (PASSED)
- "[sig-mco] MCO scale Author:sregidor-NonPreRelease-Longduration-High-52822-Create new config resources with 2.2.0 ignition boot image nodes [Disruptive] [Serial]" (PASSED)
- "[sig-mco] MCO scale Author:sregidor-NonPreRelease-Longduration-High-65923-SSH key in scaled clusters [Disruptive] [Serial]" (PASSED)
-- "[sig-mco] MCO scale Author:sregidor-NonHyperShiftHOST-NonPreRelease-Longduration-LongDuration-High-63894-Scaleup using 4.1 cloud image[Disruptive] [Serial]" (FAILED)


Test "63894-Scaleup using 4.1 cloud image" is failing.

In this test we execute these steps:
1. Create a new machineset that uses the 4.1 cloud image to create the worker nodes
2. Scale up the new machineset and create a new worker.


When we do that, the `node-valid-hostname.service` service fails with this error:

```
# systemctl status node-valid-hostname.service
× node-valid-hostname.service - Wait for a non-localhost hostname
     Loaded: loaded (/etc/systemd/system/node-valid-hostname.service; enabled; preset: enabled)
     Active: failed (Result: timeout) since Tue 2023-10-03 13:31:48 UTC; 6min ago
    Process: 1248 ExecStart=/usr/local/bin/mco-hostname --wait (code=killed, signal=TERM)
   Main PID: 1248 (code=killed, signal=TERM)
        CPU: 341ms

Oct 03 13:26:48 localhost.localdomain mco-hostname[1248]: waiting for non-localhost hostname to be assigned
Oct 03 13:26:48 localhost.localdomain systemd[1]: Starting Wait for a non-localhost hostname...
Oct 03 13:31:48 localhost.localdomain systemd[1]: node-valid-hostname.service: start operation timed out. Terminating.
Oct 03 13:31:48 localhost.localdomain systemd[1]: node-valid-hostname.service: Main process exited, code=killed, status=15/TERM
Oct 03 13:31:48 localhost.localdomain systemd[1]: node-valid-hostname.service: Failed with result 'timeout'.
Oct 03 13:31:48 localhost.localdomain systemd[1]: Failed to start Wait for a non-localhost hostname.
```

And the crio service cannot start.

We double checked that this test is currently passing in the latest accepted nightly build 4.15.0-0.nightly-2023-10-02-223206. 

We can't add the qe-approved label, I add a comment in the jira ticket linking to the journal log.

**Comment by cgwalters** [comment]
Thanks, can you attach the journal?

**Comment by sergiordlr** [comment]
This is the journal 
[journal-aws-4-1.log](https://github.com/openshift/machine-config-operator/files/12794833/journal-aws-4-1.log)

**Comment by cgwalters** [comment]
Thanks.  Hmm, strange so I see:
`Tue 2023-10-03 13:26:47 UTC localhost.localdomain NetworkManager.service[999]: <info>  [1696339607.3853] policy: set-hostname: set hostname to 'ip-10-0-21-45' (from DHCPv4)`

and yet the hostname change isn't reflected.


Digging into [this 4.14 vsphere run](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-ovn-upi/1708881452000612352) we clearly see in the journal:

```
Mar 10 17:07:52.580333 localhost.localdomain NetworkManager[1162]: <info>  [1678468072.5803] policy: set-hostname: set hostname to 'ip-10-0-146-109' (from DHCPv4)
Mar 10 17:07:52.581096 ip-10-0-146-109 systemd-hostnamed[1172]: Changed host name to 'ip-10-0-146-109'
```

So that's just weird.  

Ah wait...wait, I think I see.  The firstboot OS update did get the correct hostname set.  So...huh, did something set the static hostname on firstboot to localhost?  I see
`Tue 2023-10-03 13:26:47 UTC localhost.localdomain NetworkManager.service[999]: <info>  [1696339607.2314] hostname: static hostname changed from (none) to "localhost.localdomain"` in the subsequent boot...

**Comment by cgwalters** [comment]
@sergiordlr can you also attach the journal from the node that successfully joined *without* this change?

**Comment by sergiordlr** [comment]
I reproduced the behavior with 4.1 cloud images. I add 2 more journals. One from a machine that could be added to the cluster, and another one from a machine that could not be added to the cluster.

[journal-aws-4-1-success.log](https://github.com/openshift/machine-config-operator/files/12800650/journal-aws-4-1-success.log)   (node-valid-hostname.service)
[journal-aws-4-1-failure.log](https://github.com/openshift/machine-config-operator/files/12800653/journal-aws-4-1-failure.log)   (our fix)

It seems that the `node-valid-hostname.service` service is failing in both, but in 4.15.0-0.nightly-2023-10-02-223206 the machine can join even if this service failed.

**Comment by sergiordlr** [comment]
I have triggered the upgrade. I will post the result here once it is finished.

**Comment by sergiordlr** [comment]
The new release image cannot add the worker. This is the journal log

[jounal-aws-4-1-failed3.log](https://github.com/openshift/machine-config-operator/files/12806452/jounal-aws-4-1-failed3.log)

The image can be manually pulled from the node 
```
[core@ip-10-0-21-245 ~]$ podman pull registry.build05.ci.openshift.org/ci-ln-4illsfk/stable@sha256:416817a362a826981adef576187101b26cba064bef3dad4c2b2fb9aac98b5ee3
Trying to pull registry.build05.ci.openshift.org/ci-ln-4illsfk/stable@sha256:416817a362a826981adef576187101b26cba064bef3dad4c2b2fb9aac98b5ee3...Getting image source signatures
Copying blob 97da74cc6d8f: 13.88 MiB / 75.84 MiB [=====>-----------------------]
Copying blob d8190195889e: 0 B / 1.40 KiB [------------------------------------]
Copying blob 97da74cc6d8f: 18.82 MiB / 75.84 MiB [=======>---------------------]

```

**Comment by cgwalters** [comment]
OK I figured out what's going on, it's not technically this PR breaking things.  The problem is that `registry.build05.ci.openshift.org` is exposing images as manifest lists:

```
$ skopeo inspect --raw -n docker://registry.build05.ci.openshift.org/ci-ln-4illsfk/stable@sha256:416817a362a826981adef576187101b26cba064bef3dad4c2b2fb9aac98b5ee3 | jq .
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
```

And that's just not supported in this ancient podman.

The reason this works in production is because we haven't manifest listed the official images on quay:

```
$ skopeo inspect -n --raw docker://(oc adm release info --image-for=rhel-coreos quay.io/openshift-release-dev/ocp-release:4.14.0-rc.4-x86_64) | jq .
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
```

Now, old 4.1 era bootimages would then break if we ever did officially manifest list, which we should at some point, but hopefully by then we'll have updated bootimages.

So...to test this pre-merge I think we'd need to copy the release image to quay.io and explicitly un-manifest list it.

**Comment by cgwalters** [comment]
I've mirrored to quay.io/cgwalters/release:4.14.0-0.ci.test-2023-10-04-145907-ci-ln-4illsfk-latest-x86_64 and kicked off a cluster upgrade

**Comment by cgwalters** [comment]
Ohhhh right; actually `oc adm release mirror` doesn't *rewrite* the pull specs in the image, so that didn't help.  We'll have to manually craft a release image with just this updated MCO code instead...looking

**Comment by sergiordlr** [comment]
We have installed the fix in azure and it was installed OK. (We saw an azure installation failing because of kargs problems in a master node, this problem does not seem to be related to our fix and in a retry the installation succeeded)

We have installed the fix using in IPI on OSP and we have been able to create 10 new workers using a 4.6 cloud image.

We have installed the fix using UPI on Vsphere and we have been able to create 10 new workers using a 4.6 cloud image.

No ordering cycle problems found.

The only problem that we found happens when using 4.1 cloud images. The problem happens because the podman version is too old and the pre-merge images cannot be pulled correctly using this version. This problem should not happen with the nightly/official builds. Since we cannot test it pre-merge, what we will do is to test it post-merge using the nightly builds.

We can add the qe-approved label.

/label qe-approved

**Comment by sergiordlr** [comment]
/remove-label qe-approved

While investigating the azure error I found that maybe it is related to our fix. Please, could you have a look at this journal log?
[journal-azure-master.log](https://github.com/openshift/machine-config-operator/files/12818866/journal-azure-master.log)

**Comment by cgwalters** [comment]
Good catch!  Latest commit should fix it.

**Comment by cgwalters** [comment]
(hmm we should add a test case to the e2e that verifies there are no ordering cycles across the board)

**Comment by cgwalters** [comment]
The [e2e-azure run](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3865/pull-ci-openshift-machine-config-operator-master-e2e-azure/1709944157167947776) has no ordering cycles in the journal.

**Comment by sergiordlr** [comment]
We installed it in  UPI on GCP, IPI on GCP, UPI on Azure, IPI on Azure

In IPI on OSP
1- Install
2- Create 10 new workers using the 4.6 cloud image

In UPI on Vsphrere
1. Install
2. Create 20 new workers using the 4.6 cloud image.

No problems were found.

The only problem was the 4.1 image in test case  "63894-Scaleup using 4.1 cloud image". It will have to be tested post-merge.

We can add the qe-approeved label 

/label qe-approved

**Comment by cgwalters** [comment]
I think the current presubmit failures were transient flake
/retest

**Comment by cgwalters** [comment]
I think we're good for someone to lgtm this?

**Comment by sinnykumari** [comment]
Let's get this in. We have good signal from payload testing and our QE testing.
/lgtm

**Comment by sinnykumari** [comment]
@rioliu-rh @sergiordlr Is there still anything we need before remove hold on the PR?

**Comment by cgwalters** [comment]
Cherry pick is in https://github.com/openshift/machine-config-operator/pull/3967 - just trivial conflicts.

**Pull request merged**
